### PR TITLE
fix(sdk): Rename app start measurements to match other SDKs

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring

## :scroll: Description

Renames the `app start measurement` to `app_start_warm` and keeps the `op` as `app.start.warm` to be aligned with the other mobile SDK.

- docs: https://github.com/getsentry/sentry-docs/pull/6368

## :bulb: Motivation and Context

- https://github.com/getsentry/sentry-react-native/issues/2490

Although the mentioned issue is closed and the main cause of the long `app start time` was resolved in 

- https://github.com/getsentry/sentry-react-native/pull/2310

I believe users are seeing `app start times` only till sdk version `4.5.0`

- https://github.com/getsentry/sentry-react-native/pull/2487

as the measurement was renamed from `app_start_warm` to `app.start.warm`.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: test/tracing/reactnativetracing.test.ts [hunk 1]
```diff
@@ -114,11 +114,11 @@ describe('ReactNativeTracing', () => {
 
             expect(
               // @ts-ignore access private for test
-              transaction._measurements['app.start.cold'].value
+              transaction._measurements['app_start_cold'].value
```
Comment (marandaneto) [inline test/tracing/reactnativetracing.test.ts:117]: L: Let's create constants for this too?

**Review by marandaneto — APPROVED** [review]
Thanks @krystofwoldrich

**Comment by fabiendem** [comment]
@krystofwoldrich Are you planning on shipping this in the `4.*` version of the SDK?
Or should we consider `4.*` as no longer maintained?

Thanks for the fix

**Comment by krystofwoldrich** [comment]
@fabiendem `v4` is not actively maintained and won't get new features. It might get some critical fixes, but this likely is not one of them.

As the start-up times from the v4 can be viewed as custom measurements if your plan supports it.

**Pull request merged**